Rakefile

```diff
@@ -86,3 +86,190 @@ EOM
+    unless servicenow_group
+      servicenow_group = { 'name' => 'servicenow_nodes' }
+      inventory_hash['groups'].push(servicenow_group)
+    end
+    servicenow_group['targets'] = [{
+      'uri' => "#{on_master(&:uri)}:1080",
+      'config' => {
+        'transport' => 'remote',
+        'remote' => {
+          'user' => 'mock_user',
+          'password' => 'mock_password'
+        }
+      },
+      'vars' => {
+        'is_servicenow_instance' => true
+      }
+    }]
+    write_to_inventory_file(inventory_hash, 'inventory.yaml')
+  end
+
+  desc 'Sets up the ServiceNow CMDB with entries for each VM'
+  task :setup_servicenow_cmdb do
+    if sys_id = CMDBHelpers.record_ids[on_master(&:uri)]
+      # Double-check that the CMDB record exists on the ServiceNow instance
+      begin
+        CMDBHelpers.get_record(sys_id)
+        puts("A CMDB record's already been created for the master (sys_id = #{sys_id})")
+        next
+      rescue CMDBRecordRetrievalError => e
+        raise "inventory.yaml reports a sys_id of #{sys_id} for master's CMDB record, but servicenow_tasks::get_record fails to retrieve the record: #{e}"
+      end
+    end
+
+    # CMDB record doesn't exist so create it
+    puts("Creating the master's CMDB record ...")
+    fields_template = JSON.parse(File.read('spec/support/acceptance/cmdb_record_template.json'))
+    fields = CMDBHelpers.create_record(fields_template.merge('fqdn' => on_master(&:uri)))
+    sys_id = fields['sys_id']
+
+    # Add the CMDB record to the inventory.yaml file
+    puts("Adding the created CMDB record (sys_id = #{sys_id}) to inventory['vars'] ...")
+    master_uri = on_master(&:uri)
+    inventory_hash = inventory_hash_from_inventory_file
+    inventory_hash['vars'] = {
+      'cmdb_record_ids' => CMDBHelpers.record_ids.merge(master_uri => sys_id),
+    }
+    write_to_inventory_file(inventory_hash, 'inventory.yaml')
+  end
+
+  desc 'Installs the module on the master'
+  task :install_module do
+    on_master { |master| Rake::Task['litmus:install_module'].invoke(master.uri) }
+  end
+
+  desc 'Set up the test infrastructure'
+  task :setup do
+    tasks = [
+      :provision_vms,
+      :setup_pe,
+      :setup_servicenow_instance,
+      :setup_servicenow_cmdb,
+      :install_module,
+    ]
+
+    tasks.each do |task|
+      task = "acceptance:#{task}"
+      puts("Invoking #{task}")
+      Rake::Task[task].invoke
+      puts("")
+    end
+  end
+
+  desc 'Runs the tests'
+  task :run_tests do
+    puts("Running the tests ...\n")
+
+    # TODO: We can't use litmus:acceptance:parallel here b/c that will try
+    # to run the spec tests on _all_ targets in the inventory.yaml file,
+    # including our ServiceNow instance. Until litmus:acceptance:parallel
+    # has a way for you to pass-in specific targets to run the tests on,
+    # we'll just run these tests via rspec
+    system({ 'TARGET_HOST' => on_master(&:uri) }, 'bundle exec rspec ./spec/acceptance')
```

Comment:
Also per the convo with David, we should probably just run `rspec` here. The `TARGET_HOST` doesn't make sense for our multi-node test environment.

Reply:
Is a regular call to `system` here going to return all of the output streams properly? Or would it be better to call Open3 so you can examine stderr vs stdout and return the right thing?

Reply:
It outputs the right thing at the right streams. But it looks like `system` returns true if exit code was 0, false otherwise so this current code will exit 0 even if rspec fails. Will fix that.

Reply:
Updated.